auth/src/main/java/com/firebase/ui/auth/ui/phone/PhoneNumberVerificationHandler.java

```diff
@@ -58,7 +60,11 @@ public void onCodeSent(@NonNull String verificationId,
         if (force) {
             optionsBuilder.setForceResendingToken(mForceResendingToken);
         }
-        PhoneAuthProvider.verifyPhoneNumber(optionsBuilder.build());
+        if (isBrowserAvailable(activity)) {
+            PhoneAuthProvider.verifyPhoneNumber(optionsBuilder.build());
+        } else {
+            setResult(Resource.forFailure(new ActivityNotFoundException("No browser was found in this device")));
```

Comment:
Is it only logging the message, or is it also displaying something to the user?

Reply:
User sees a "Unknown error occurred" red message on the screen.
I know it's not a very explanatory message, but unfortunately the library was designed like that. We have lots of non-explanatory messages...
My favorite one is when the logcat shows `DEVELOPER_ERROR`:
https://github.com/firebase/FirebaseUI-Android/blob/4455f411e3605743bb4511d1a75cda7f398bc399/auth/src/main/java/com/firebase/ui/auth/ErrorCodes.java#L27-L29

Reply:
Indeed, this one might be the best non-explanatory message I've seen...